### ISSUE TITLE
HighestWeightModule function in Lie Algebras package

### ISSUE THREAD

**Issue by bendelc** [gap-system/gap#2491]
The HighestWeightModule(-) is giving an unexpected error message for certain weights:

### Observed behaviour

First, creating a Lie algebra works fine:
```
gap> L := SimpleLieAlgebra("G",2,Rationals);
<Lie algebra of dimension 14 over Rationals>
```
Then the following command gives the subsequent error message:
```
gap> W := HighestWeightModule(L,[4,1]);
Error, no method found! For debugging hints type ?Recovery from NoMethodFound
Error, no 1st choice method found for `[]' on 2 arguments
The 2nd argument is 'fail' which might point to an earlier problem
 at /proc/cygdrive/C/gap-4.9.1/lib/methsel2.g:250 called from
o[pos] at /proc/cygdrive/C/gap-4.9.1/lib/lierep.gi:3285 called from
lexord( novar, y, x ) at /proc/cygdrive/C/gap-4.9.1/lib/lierep.gi:3569 called from
Sort( sortmn[j], function ( x, y )
      return lexord( novar, y, x );
  end ); at /proc/cygdrive/C/gap-4.9.1/lib/lierep.gi:3569 called from
<function "unknown">( <arguments> )
 called from read-eval loop at *stdin*:3
you can 'quit;' to quit to outer loop, or
you can 'return;' to continue
brk>
```
Replacing the weight [4,1] with other "large" weights, gives a similar error message.

### Expected behaviour

For a smaller weight, one gets a result such as the following:
```
gap> V := HighestWeightModule(L,[2,2]);
<729-dimensional left-module over <Lie algebra of dimension 14 over Rationals>>
```
It is possible that there is some sort of sort of memory issue  as the dimension of the module gets larger, but one would expect a more clear memory error then.

### Copy and paste GAP banner (to tell us about your setup)

The same issue results in both a 32-bit and 64-bit installation on a Windows laptop:
```
 GAP 4.9.1 of 05-May-2018
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: i686-pc-cygwin-default32
 Configuration:  gmp 6.0.0, readline
 Loading the library and packages ...
 Packages:   AClib 1.3, Alnuth 3.1.0, AtlasRep 1.5.1, AutPGrp 1.9, Browse 1.8.7, CRISP 1.4.4, Cryst 4.1.17,
             CrystCat 1.1.8, CTblLib 1.2.2, FactInt 1.6.2, FGA 1.4.0, GAPDoc 1.6.1, IO 4.5.1, IRREDSOL 1.4,
             LAGUNA 3.9.0, Polenta 1.3.8, Polycyclic 2.13.1, PrimGrp 3.3.1, RadiRoot 2.8, ResClasses 4.7.1,
             SmallGrp 1.3, Sophus 1.24, SpinSym 1.5, TomLib 1.2.6, TransGrp 2.0.2, utils 0.54
```
```
 GAP 4.9.1 of 05-May-2018
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: x86_64-unknown-cygwin-default64
 Configuration:  gmp 6.1.0, readline
 Loading the library and packages ...
 Packages:   AClib 1.3, Alnuth 3.1.0, AtlasRep 1.5.1, AutPGrp 1.9, Browse 1.8.7, CRISP 1.4.4, Cryst 4.1.17,
             CrystCat 1.1.8, CTblLib 1.2.2, FactInt 1.6.2, FGA 1.4.0, GAPDoc 1.6.1, IO 4.5.1, IRREDSOL 1.4,
             LAGUNA 3.9.0, Polenta 1.3.8, Polycyclic 2.13.1, PrimGrp 3.3.1, RadiRoot 2.8, ResClasses 4.7.1,
             SmallGrp 1.3, Sophus 1.24, SpinSym 1.5, TomLib 1.2.6, TransGrp 2.0.2, utils 0.54
```
A colleagues has also tried the code on a Macbook, with the same result.

**Comment by bendelc** [comment]
Additional Comment: I tried installing an older version - 4.8.10:
  GAP 4.8.10, 15-Jan-2018, build of 2018-01-15 14:36:25 (GMTST)
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: i686-pc-cygwin-gcc-default32
 Libs used:  gmp, readline
 Loading the library and packages ...
 Components: trans 1.0, prim 2.1, small* 1.0, id* 1.0
 Packages:   AClib 1.2, Alnuth 3.1.0, AtlasRep 1.5.1, AutPGrp 1.8, Browse 1.8.7, CRISP 1.4.4, Cryst 4.1.13,
             CrystCat 1.1.6, CTblLib 1.2.2, FactInt 1.6.0, FGA 1.3.1, GAPDoc 1.6.1, IO 4.5.1, IRREDSOL 1.4,
             LAGUNA 3.8.0, Polenta 1.3.8, Polycyclic 2.11, RadiRoot 2.7, ResClasses 4.7.1, Sophus 1.23, SpinSym 1.5,
             TomLib 1.2.6, utils 0.49

That version is working correctly. One gets the desired output:
gap> L := SimpleLieAlgebra("G",2,Rationals);
<Lie algebra of dimension 14 over Rationals>
gap> V := HighestWeightModule(L,[4,1]);
<924-dimensional left-module over <Lie algebra of dimension 14 over Rationals>>

**Comment by fingolfin** [comment]
@bendelc Thank you very much for your detailed report, which was quite helpful in tracking down the issue; and my apologies for replying so late -- somehow, this issue got lost in the shuffle.

Anyway, on the upside, we have a fix, see PR #2617. I don't know if it will make it in GAP 4.9.2 (which is due any day now), if not, it'll be in GAP 4.9.3 (and of course in 4.10).
